### PR TITLE
Add component metadata YAML for Azure Event Hubs pubsub

### PR DESCRIPTION
**Pull request by yash-nisar**
# Description

Add component metadata YAML for Azure Event Hubs pubsub

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pubsub/azure/eventhubs/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  - name: partitionCount
+    description: "Number of partitions for the new Event Hub namespace. Used only when entity management is enabled."
+    type: number
+    default: "1"
+    example: "2"
+  - name: messageRetentionInDays
+    description: "Number of days to retain messages for in the newly created Event Hub namespace. Used only when entity management is enabled."
+    type: number
+    default: "1"
+    example: "90"
```
Comment (tmacam) [inline pubsub/azure/eventhubs/metadata.yaml:74]: If they require Entity Management and Entity Mangement can only be enabled when using Azure AD, we should move those to `azuread` auth profile. A similar comment was made to my PR on https://github.com/dapr/components-contrib/pull/2784

File: pubsub/azure/eventhubs/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        required: true
+        sensitive: true
+        description: "Connection string for the Event Hub or the Event Hub namespace. Mutally exclusive with eventHubNamespace field."
+        example: '"Endpoint=sb://{EventHubNamespace}.servicebus.windows.net/;SharedAccessKeyName={PolicyName};SharedAccessKey={Key};EntityPath={EventHub}"
+        or "Endpoint=sb://{EventHubNamespace}.servicebus.windows.net/;SharedAccessKeyName={PolicyName};SharedAccessKey={Key}"'
+builtinAuthenticationProfiles:
+  - name: "azuread"
+    metadata:
+      - name: eventHubNamespace
+        description: "The Event Hub Namespace name. Mutually exclusive with connectionString field."
```
Comment (tmacam) [inline pubsub/azure/eventhubs/metadata.yaml:25]: `connectionString` is in another auth profile so this is implicit. 

```suggestion
        description: "The Event Hub Namespace name. "
```

**Review by tmacam — CHANGES_REQUESTED** [review]
Forwarding some of the changes I got for https://github.com/dapr/components-contrib/pull/2784

File: pubsub/azure/eventhubs/metadata.yaml (hunk 2 above)
Comment (yash-nisar) [reply pubsub/azure/eventhubs/metadata.yaml]: Addressed

File: pubsub/azure/eventhubs/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        or "Endpoint=sb://{EventHubNamespace}.servicebus.windows.net/;SharedAccessKeyName={PolicyName};SharedAccessKey={Key}"'
+builtinAuthenticationProfiles:
+  - name: "azuread"
+    metadata:
+      - name: eventHubNamespace
+        description: "The Event Hub Namespace name."
+        example: '"namespace"'
+        required: true
+      - name: enableEntityManagement
+        description: "Boolean value to allow management of the EventHub namespace and storage account."
```
Comment (ItalyPaleAle) [inline pubsub/azure/eventhubs/metadata.yaml:29]:
```suggestion
        description: "Allow management of the Event Hub namespace and storage account."
```

File: pubsub/azure/eventhubs/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        example: "false"
+
+      # The following four properties are needed only if enableEntityManagement is set to true
+      - name: resourceGroupName
+        description: "Name of the resource group the Event Hub namespace is part of. Required when entity management is enabled."
+        type: string
+        required: false
+        example: "test-rg"
+      - name: subscriptionID
+        description: "Azure subscription ID value. Required when entity management is enabled."
```
Comment (ItalyPaleAle) [inline pubsub/azure/eventhubs/metadata.yaml:42]:
```suggestion
        description: |
          Azure subscription ID value. Required when entity management is enabled
        example: '"00112233-4455-6677-8899-aabbccddeeff"'
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Could you please copy the descriptions and examples for matching metadata fields from https://github.com/yash-nisar/components-contrib/blob/17d0a8a9708d8466e82a6e74110d7c28e508d3b5/bindings/azure/eventhubs/metadata.yaml ?

**Review by tmacam — CHANGES_REQUESTED** [review]
LGTM if the changes requested by Ale are implemented

**Review by berndverst — APPROVED** [review]

**Comment by yash-nisar** [comment]
@yaron2 I believe you assigned this PR to me by mistake so I'm unassigning myself for now.

**Pull request merged**